app/src/androidTest/java/at/tacticaldevc/oat/utils/PrefsDataTest.java

```diff
@@ -2,57 +2,241 @@
+
+import static at.tacticaldevc.oat.utils.Prefs.addNewOnStartupPermissionRequest;
 import static at.tacticaldevc.oat.utils.Prefs.addTrustedContact;
-import static at.tacticaldevc.oat.utils.Prefs.getAllTrustedContacts;
+import static at.tacticaldevc.oat.utils.Prefs.fetchCommandTriggerWord;
+import static at.tacticaldevc.oat.utils.Prefs.fetchOnStartupPermissionRequests;
+import static at.tacticaldevc.oat.utils.Prefs.fetchTrustedContacts;
+import static at.tacticaldevc.oat.utils.Prefs.removeOnStartupPermissionRequest;
 import static at.tacticaldevc.oat.utils.Prefs.removeTrustedContact;
+import static at.tacticaldevc.oat.utils.Prefs.saveCommandTriggerWord;
+import static at.tacticaldevc.oat.utils.Prefs.savePassword;
+import static at.tacticaldevc.oat.utils.Prefs.verifyApplicationPassword;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 
 public class PrefsDataTest {
     private static final String DOCUMENT_NAME_TEST = "oat-data";
     private final static String KEY_TRUSTED_CONTACTS = "trusted-contacts";
+    private final static String KEY_COMMAND_PASSWORD = "password";
+    private final static String KEY_COMMAND_PASSWORD_SALT = "pwdsalt";
+    private final static String KEY_COMMAND_TRIGGER = "cmd-trigger";
+    private final static String KEY_MISSING_PERMISSIONS_TO_REQUEST_ON_STARTUP = "missing-permission";
 
     @Before
     public void init() {
         clean();
     }
 
+    // password
+    @Test
+    public void savePasswordWithInvalidValues() {
+        // test
+        assertThrows(IllegalArgumentException.class, () -> savePassword(null, null));
+        assertThrows(IllegalArgumentException.class, () -> savePassword(InstrumentationRegistry.getInstrumentation().getTargetContext(), null));
+        assertThrows(IllegalArgumentException.class, () -> savePassword(null, "new Password"));
+        assertThrows(IllegalArgumentException.class, () -> savePassword(null, ""));
+        assertThrows(IllegalArgumentException.class, () -> savePassword(null, "\n\r"));
+    }
+
+    @Test
+    public void savePasswordWithoutData() {
+        // prepare
+        String newPassword = "NewPassword";
+
+        // test
+        savePassword(InstrumentationRegistry.getInstrumentation().getTargetContext(), newPassword);
+
+        // assert
+        SharedPreferences prefs = InstrumentationRegistry.getInstrumentation().getTargetContext().getSharedPreferences(DOCUMENT_NAME_TEST, Context.MODE_PRIVATE);
+        String psswdHash = prefs.getString(KEY_COMMAND_PASSWORD, null);
+        byte[] salt = Base64.decode(prefs.getString(KEY_COMMAND_PASSWORD_SALT, null), Base64.NO_WRAP);
+        MessageDigest algorithm;
+        try {
+            algorithm = MessageDigest.getInstance("SHA-256");
+            algorithm.update(salt);
+            String expectedHash = Base64.encodeToString(algorithm.digest(newPassword.getBytes()), Base64.NO_WRAP);
```

Comment:
Calculating the expected hash with the calculated hash will always be true. Wouldn't it make more sense if you would hard-code it?

Reply:
Nvm made a mistake